src/NotificationEventAbstract.php

```diff
@@ -94,6 +94,18 @@ public static function raise(
                  $notificationtarget->addForTarget($target, $options);
 
                 foreach ($notificationtarget->getTargets() as $users_infos) {
+                    $user = new User();
+                    //do not notify if user explicitly refused it
+                    //Except for CommonITILOject (already manage by use_notification)
+                    if (
+                        !is_a($item, CommonITILObject::class)
+                        && isset($users_infos['users_id'])
+                        && $user->getFromDB($users_infos['users_id'])
+                        && !$user->isUserNotificationEnable()
+                    ) {
+                        continue;
+                    }
+
```

Comment:
Some notifications should not be blocked, for instance, `passwordexpires` and `passwordforget` notifications should never be blocked.

Maybe the solution is to handle this using the existing `NotificationTarget::validateSendTo()` method, instead of hardcoding it here.
```suggestion
```

Reply:
done